src/Exception/NewOrderWrongTotalException.php

```diff
@@ -0,0 +1,13 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Setono\SyliusOrderEditPlugin\Exception;
+
+final class NewOrderWrongTotalException extends \RuntimeException
+{
+    public static function occur(): self
```

Comment:
Is this some kind of pattern?

Not sure what it improves over `new NewOrderWrongTotalException`.

If I do something like this, I would do something like this instead:

```php
public static function assert(int $oldTotal, int $newTotal): void
{
    if($oldTotal < $newTotal) {
        throw new self('New order total is greater than the initial order total');
    }
}
```

Reply:
A matter of taste, I believe 🖖 Can for sure switch to `new ...` if you prefer. Regarding the if in exception, I would like to keep it in a service that is responsible for the application logic.

Reply:
Yes, totally agree. I just never saw this before